openmc/weight_windows.py

```diff
@@ -125,7 +125,9 @@ def __init__(
         self.mesh = mesh
         self.id = id
         self.particle_type = particle_type
-        self.energy_bounds = energy_bounds
+        self._energy_bounds = None
```

Comment:
It would also be nice if `None` is a valid value of `_energy_bounds` then it should be able to be set through the property setter. That way we don't have to set the private variable in the constructor or do the `is not None` check.

Reply:
> It would also be nice if None is a valid value of _energy_bounds then it should be able to be set through the property setter. That way we don't have to set the private variable in the constructor or do the is not None check.

Agreed. That would be nice. Though I'd say this is true of other attributes in the Python API ([example](https://github.com/openmc-dev/openmc/blob/36f229eb01de0a8cb7d482d9a9b7a2f2c615f3f5/openmc/source.py#L185)) though and is probably something that should change across the board if we want to tackle that.

Reply:
sounds good